lib/cudnn/convolution.jl

```diff
@@ -182,7 +182,7 @@ function cudnnConvolutionFwdAlgoPerf(xDesc, x, wDesc, w, convDesc, yDesc, y, bia
         perfResults = Array{cudnnConvolutionFwdAlgoPerf_t}(undef,requestedAlgoCount)
         workspaceSize() = cudnnFindConvolutionAlgorithmWorkspaceSize(x)
         with_workspace(workspaceSize) do workspace
-            cudnnFindConvolutionForwardAlgorithmEx(handle(),xDesc,x,wDesc,w,convDesc,yDesc,y,requestedAlgoCount,returnedAlgoCount,perfResults,workspace,sizeof(workspace))
+            cudnnFindConvolutionForwardAlgorithmEx(handle(),xDesc,x,wDesc,w,convDesc,yDesc,similar(y),requestedAlgoCount,returnedAlgoCount,perfResults,workspace,sizeof(workspace))
```

Comment:
Following from https://github.com/FluxML/NNlibCUDA.jl/issues/37#issuecomment-1146189142, perhaps an optional arg could be added to `cudnnConvolutionFwdAlgoPerf` which controls whether a `y`/`dx` are re-used as output buffers. Making it false by default would be fine. `cudnnConvolutionForwardAD` would be responsible for flipping it to true when `beta` is zero.

Reply:
I was first hesitant, but I just found the proof that this can be done safely in the cudnn docs [here](https://docs.nvidia.com/deeplearning/cudnn/developer-guide/index.html#scaling-parameters), stating "When beta is zero, the output is not read and may contain uninitialized data (including NaN)."

Reply:
Ok, please check again if this is what you meant :)